testsuite/testsuite_impl/my_advanced_test/my_advanced_test_.go

```diff
@@ -0,0 +1,528 @@
+
+	key, err := getPrivateKey(serviceCtx)
+	if err != nil {
+		return stacktrace.Propagate(err, "Failed to get private key")
+	}
+
+	transactor, err := bind.NewKeyedTransactorWithChainID(key.PrivateKey, big.NewInt(15))
+	if err != nil {
+		log.Fatalf("Failed to create authorized transactor: %v", err)
+	}
+	transactor.GasPrice = big.NewInt(5)
+	address, tx, helloWorld, err := bindings.DeployHelloWorld(transactor, gethClient)
+	if err != nil {
+		return stacktrace.Propagate(err, "An error occurred deploying the HelloWorld contract on the ETH Network")
+	}
+	fmt.Printf("Contract pending deploy: 0x%x\n", address)
+	fmt.Printf("Transaction waiting to be mined: 0x%x\n\n", tx.Hash())
+
+	if err := waitUntilTransactionMined(gethClient, tx.Hash()); err != nil {
+		return stacktrace.Propagate(err, "An error occurred waiting for the HelloWorld contract to be mined")
+	}
+	logrus.Info("Deployed Hello World contract")
+
+	name, err := helloWorld.Greet(&bind.CallOpts{Pending: true})
+	if err != nil {
+		log.Fatalf("Failed to retrieve pending name: %v", err)
+	}
+	fmt.Println("Pending name:", name)
+
+	listAccountsCmd := []string{
+		"/bin/sh",
+		"-c",
+		fmt.Sprintf("geth attach data/geth.ipc --exec eth.accounts"),
+	}
+
+	exitCode, logOutput, err := serviceCtx.ExecCommand(listAccountsCmd)
+	if err != nil {
+		return stacktrace.Propagate(err, "Executing command returned an error with logs: %+v", string(*logOutput))
+	}
+	if exitCode != execCommandSuccessExitCode {
+		return stacktrace.NewError("Executing command returned an failing exit code with logs: %+v", string(*logOutput))
+	}
+
+	if err = validateAccount(string(*logOutput)); err != nil {
+		return stacktrace.Propagate(err, "Validate account error")
+	}
+
+	return nil
+}
+
+// ====================================================================================================
+//                                       Private helper functions
+// ====================================================================================================
+func startEthBootnode(networkCtx *networks.NetworkContext) (string, error) {
+	containerCreationConfig, runConfigFunc := getEthereumServiceConfigurationsForBootnode()
+
+	_, hostPortBindings, err := networkCtx.AddService(bootnodeID, containerCreationConfig, runConfigFunc)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred adding the Ethereum Bootnode")
+	}
+
+	logrus.Infof("Added Ethereum Bootnode service with host port bindings: %+v ", hostPortBindings)
+
+	serviceCtx, err := networkCtx.GetServiceContext(bootnodeID)
+	if err != nil {
+		return "", stacktrace.Propagate(err, "An error occurred getting the Ethereum Bootnode service context")
+	}
+	logrus.Infof("Got service context for Ethereum Bootnode service '%v' and IP address '%v", serviceCtx.GetServiceID(), serviceCtx.GetIPAddress())
+
+	if err := waitForStartup(serviceCtx.GetIPAddress(), gethTimeBetweenIsAvailablePolls, gethMaxIsAvailablePolls); err != nil {
```

Comment:
I didn't use the method `WaitForEndpointAvailability` from Kurt Core because it only executes HTTP Get requests and it needs the HTTP Post method for RPC calls.